core-plugins/src/main/java/co/cask/hydrator/plugin/realtime/sink/RealtimeKinesisStreamSink.java

```diff
@@ -0,0 +1,199 @@
+  private final KinesisConfig config;
+  private static AmazonKinesisClient kinesisClient;
+
+
+  public RealtimeKinesisStreamSink(KinesisConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.name),
+                                "Stream name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessKey),
+                                "Access Key required");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.awsAccessSecret),
+                                "Access Key secret required");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.partition),
+                                "partition name should be non-null, non-empty.");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(config.bodyField),
+                                "Please map the input field containing data");
+
+  }
+
+  @Override
+  public void initialize(RealtimeContext context) throws Exception {
+    super.initialize(context);
+    BasicAWSCredentials awsCred = new BasicAWSCredentials(config.awsAccessKey, config.awsAccessSecret);
+    kinesisClient = new AmazonKinesisClient(awsCred);
+    createAndWaitForStreamToBecomeAvailable(kinesisClient,
+                                            config.name,
+                                            config.shardCount);
+  }
+
+  @Override
+  public int write(Iterable<StructuredRecord> structuredRecords, DataWriter dataWriter) throws Exception {
+    int numRecordsWritten = 0;
+    for (StructuredRecord structuredRecord : structuredRecords) {
+      Schema schema = structuredRecord.getSchema();
+      Object data = structuredRecord.get(config.bodyField);
+      if (data == null) {
+        LOG.debug("Found null data. Skipping record.");
+        continue;
+      }
+      Schema.Field dataSchemaField = schema.getField(config.bodyField);
+      PutRecordRequest putRecordRequest = new PutRecordRequest();
+      putRecordRequest.setStreamName(config.name);
+      putRecordRequest.setPartitionKey(config.partition);
+      switch (dataSchemaField.getSchema().getType()) {
+        case BYTES:
+          numRecordsWritten += writeBytes(putRecordRequest, data);
+          break;
+        case STRING:
+          numRecordsWritten += writeString(putRecordRequest, data);
+          break;
+        default:
+          if (dataSchemaField.getSchema().getType().isSimpleType()) {
+            numRecordsWritten += writeBytes(putRecordRequest, String.valueOf(data));
+            break;
+          }
+          LOG.debug("Type {} is not supported for writing to stream", data.getClass().getName());
+          break;
+      }
+    }
+    return numRecordsWritten;
+  }
+
+  private int writeString(PutRecordRequest putRecordRequest, Object data) {
```

Comment:
@override write method inherited from realtime sink, is expected to return the number of records written successfully. These helper write methods are returning ints to increment the numRecordsWritten in write method.